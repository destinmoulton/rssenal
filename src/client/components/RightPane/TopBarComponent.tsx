import * as React from "react";

import SortMenu from "./SortMenu";
import SettingsModalContainer from "../../containers/Modals/SettingsModalContainer";

interface ITopBarProps {
    title: string;
    onChangeSort: (e: any, data: any) => void;
    sortBy: string;
}

class TopBarComponent extends React.Component<ITopBarProps> {
    render() {
        const { onChangeSort, sortBy, title } = this.props;
        return (
            <div className="rss-rightpane-topbar">
                <div className="rss-rightpane-topbar-title-container">
                    {title}
                </div>
                <div className="rss-rightpane-topbar-sortselect-container">
                    <SortMenu onChange={onChangeSort} currentSortBy={sortBy} />&nbsp;&nbsp;
                    <SettingsModalContainer />
                </div>
            </div>
        );
    }
}

export default TopBarComponent;
