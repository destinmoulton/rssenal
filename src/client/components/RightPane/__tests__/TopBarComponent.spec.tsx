import * as React from "react";
import { shallow } from "enzyme";

import TopBarComponent from "../TopBarComponent";

describe("<TopBarComponent />", () => {
    it("matches the snapshot", () => {
        const title = "Test Title";
        const onChangeMock = jest.fn();
        const sortBy = "sort:byTest";
        const wrapper = shallow(
            <TopBarComponent
                title={title}
                onChangeSort={onChangeMock}
                sortBy={sortBy}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
