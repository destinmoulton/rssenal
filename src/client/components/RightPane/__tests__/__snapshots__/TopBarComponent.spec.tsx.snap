// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBarComponent /> matches the snapshot 1`] = `
<div
  className="rss-rightpane-topbar"
>
  <div
    className="rss-rightpane-topbar-title-container"
  >
    Test Title
  </div>
  <div
    className="rss-rightpane-topbar-sortselect-container"
  >
    <SortMenu
      currentSortBy="sort:byTest"
      onChange={[MockFunction]}
    />
      
    <Connect(SettingsModalComponent) />
  </div>
</div>
`;
